### PR TITLE
refactor: clean up the CLI context arg

### PR DESCRIPTION
**Pull request by manekinekko**
Adding @alexweininger since this change may affect VS Code extension.

### CODE REVIEW

File: src/core/utils/cli-config.ts [hunk 1]
```diff
@@ -19,9 +19,9 @@ export const getCurrentSwaCliConfigFromFile = () => currentSwaCliConfigFromFile;
 export const swaCliConfigFileExists = (configFilePath: string) => existsSync(configFilePath);
 
 export async function getConfigFileOptions(
-  contextOrConfigEntry: string | undefined,
+  outputLocationOrConfigEntry: string | undefined,
```
Comment (sinedied) [inline src/core/utils/cli-config.ts:22]: I think we should keep context internal in this file, as using outputLocation instead makes it more confusing for commands that don't include one (ie login/init but also upcoming build link etc).

**Review by sinedied — CHANGES_REQUESTED** [review]

File: src/cli/commands/deploy.ts [hunk 2]
```diff
@@ -175,21 +186,20 @@ export async function deploy(deployContext: string, options: SWACLIConfig) {
   }
   logger.log(``);
 
-  if (options.printToken) {
+  if (printToken) {
```
[the remark this answers is not in the document]
Comment (manekinekko) [reply src/cli/commands/deploy.ts:189]: this option is only used to print the token and exit.

File: src/core/utils/cli-config.ts (hunk 1 above)
Comment (manekinekko) [reply src/core/utils/cli-config.ts:22]: Why rename `contextOrConfigEntry` to `outputLocationOrConfigEntry`?

Currently, this is how `getConfigFileOptions(a, b)` can be invoked:
- `swa login` : `a` is undefined => we skip loading any config
- `swa init name`:  `a` is undefined => we skip loading any config
- `swa start ./folder`:  context is outputLocation
- `swa start http://localhost:3000`: context is outputLocation (aka devServerUrl)
- `swa start name`: context is configEntry
- `swa deploy ./folder`:  context is outputLocation
- `swa deploy name`:  context is configEntry

File: src/cli/commands/deploy.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sinedied) [reply src/cli/commands/deploy.ts:189]: I missed the exit part, my bad!

**Review by sinedied — APPROVED** [review]

**Pull request merged**